bench/wtperf/wtperf.c

```diff
@@ -1411,21 +1411,17 @@ flush_tier_worker(void *arg)
             break;
         /*
          * In order to get all the data into the last object when the work is done, we need to call
-         * checkpoint to get all the data into this object before calling flush_tier.
+         * checkpoint with flush_tier enabled to get all the data into this object.
          */
+		wtperf->flush = true;
         if (stop) {
             lprintf(wtperf, 0, 1, "Last call before stopping flush_tier");
-            if ((ret = session->checkpoint(session, NULL)) != 0) {
+            if ((ret = session->checkpoint(session, "flush_tier=(enabled)")) != 0) {
                 lprintf(wtperf, ret, 0, "Checkpoint failed.");
                 goto err;
             }
         }
```

Comment:
@ddanderson , wtperf also has a separate checkpoint worker that does checkpoints at regular intervals and also performs a final checkpoint on stop if tiered storage is enabled. It seems to be that we need to remove the flush worker altogether and just run the checkpoint worker with flush_tier=(enabled) whenever tiered storage is enabled. WDYT? 

@sueloverso, what are your thoughts on this change?

Reply:
One problem I see with eliminating the flush_tier_worker is that we would not be able to control separately the flush interval and the checkpoint interval. I'll keep it in for now, but it's worth discussing whether we should get rid of it.

Reply:
Right, it can be done either way.  The easiest is to leave two threads in place.  We could also require the flush_tier intervals to be multiple of checkpoint intervals, and then have a simple way to have a single thread.  There's also a function `testutil_tiered_sleep` that is designed to solve this problem, but it requires that the test application use the TEST_OPTS structure, not sure if that makes things more difficult or not.